rust/src/core.rs

```diff
@@ -28,6 +28,10 @@ use crate::flow::Flow;
 pub enum DetectEngineState {}
 pub enum AppLayerDecoderEvents {}
 pub enum GenericVar {}
+#[repr(C)]
+pub struct DetectEngineThreadCtx {
```

Comment:
this uses a different logic that the forward declarations above it. Any reason for this?

Reply:
Quoting https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs

> Notice that it is a really bad idea to use an empty enum as FFI type.

Reply:
And https://github.com/OISF/suricata/pull/13085 does a better job for `DetectEngineThreadCtx` bindgening it, and bindgen does that scheme see https://github.com/OISF/suricata/pull/13085/files#diff-25a0044cb0cfaafb69e9241adddbbabf73f8448b23538e07bba248e4f40f056aR6

So, you can merge https://github.com/OISF/suricata/pull/13085 (or its successor) first and I will rebase this (or the other way around but in the end, I will just keep the bindgen'd `DetectEngineThreadCtx` )

Reply:
Actually, both PRs handled `DetectEngineThreadCtx` and `DetectEngineCtx` so no conflict yet